libs/pdi-scanner/src/main.rs

```diff
@@ -178,111 +179,123 @@ fn main() -> color_eyre::Result<()> {
 
     let mut client: Option<Client<Scanner>> = None;
     let mut raw_image_data = RawImageData::new();
+    let mut scan_in_progress = false;
 
     // Main loop:
     // - Process any commands received on stdin
     // - Process any events or image data received from the scanner
     loop {
         match stdin_rx.try_recv() {
-            Ok(command) => match (&mut client, command) {
-                (_, Command::Exit) => {
+            Ok(command) => {
+                if command == Command::Exit {
```

Comment:
You can avoid implementing `PartialEq` with this:
```suggestion
                if matches!(command, Command::Exit) {
```

Reply:
thanks!